src/main/kotlin/org/rust/lang/core/types/infer/PatternMatching.kt

```diff
@@ -126,6 +125,36 @@ fun RsPat.extractBindings(fcx: RsTypeInferenceWalker, type: Ty, defBm: RsBinding
     }
 }
 
+private fun inferSlicePatsTypes(
+    fcx: RsTypeInferenceWalker,
+    patList: List<RsPat>,
+    bm: RsBindingModeKind,
+    sliceType: Ty
+) {
+    fun calcRestSize(arrayTy: TyArray): Const {
+        val arraySize = arrayTy.size ?: return CtUnknown
+
+        val patRestCount = patList.count { it is RsPatRest || it is RsPatIdent && it.pat is RsPatRest }
```

Comment:
Let's extract `it is RsPatRest || it is RsPatIdent && it.pat is RsPatRest` into some extension method/function because you use it at least two times